x-pack/plugins/secops/public/components/range_date_picker/index.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiDatePickerRange, EuiFlexGroup, EuiFlexItem, EuiFormControlLayout } from '@elastic/eui';
+import { get, getOr } from 'lodash/fp';
+import moment, { Moment } from 'moment';
+import React from 'react';
+import { connect } from 'react-redux';
+import { inputsActions, inputsModel, State } from '../../store';
+import { GlobalDateButton } from './global_date_button';
+import { QuickSelectPopover } from './quick_select_popover';
+import { UpdateButton } from './update_button';
+
+export type DateType = 'relative' | 'absolute';
+
+interface RangeDatePickerStateRedux {
+  from: number;
+  to: number;
+  isTimerOn: boolean;
+  duration: number;
+  isLoading: boolean;
+  refetch: inputsModel.Refetch[];
+}
+
+interface RangeDatePickerDispatchProps {
+  setAbsoluteRangeDatePicker: (params: { id: string; from: number; to: number }) => void;
+  setRelativeRangeDatePicker: (
+    params: { id: string; option: string; from: number; to: number }
+  ) => void;
+  startAutoReload: (params: { id: string }) => void;
+  stopAutoReload: (params: { id: string }) => void;
+  setDuration: (params: { id: string; duration: number }) => void;
+}
+interface OwnProps {
+  id: string;
+  disabled?: boolean;
+}
+
+type RangeDatePickerProps = OwnProps & RangeDatePickerDispatchProps & RangeDatePickerStateRedux;
+
+export interface RecentlyUsedI {
+  kind: string;
+  text: string | number[];
```

Comment:
Wait a second .... :-)

I think you're playing a bit of a trick here with these two types and type coercion no?

Are you making it so that you can have a string but later change that string into an array of numbers automatically? Something seems off with this. I would choose something like an array of numbers and then change it into a string or the other way around when it needs to fit into a new type.

As-is, this is a bit confusing to me and I think that Typescript is just allowing through its grace 1 type or the other and using java script's coercion to automatically do its job (which could be error prone later)